### ISSUE TITLE
`poetry init` does not allow `é` in the authors field

### ISSUE THREAD

**Issue by jmg-duarte** [python-poetry/poetry#8779]
- **Poetry version**: 1.7.1
- **Python version**: 3.12.0
- **OS version and name**: MacOS Sonoma 14.1.1
- **pyproject.toml**: NA

- [X] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a [recommended method](https://python-poetry.org/docs/#installation).
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

I've read https://github.com/python-poetry/poetry/issues/798 and https://github.com/python-poetry/poetry/issues/4258, those issues deal with "post-`init`", my issue is _with_/_about_ `init`.

As stated in the title, `poetry init` does not allow for `é` to exist in the authors field.
```
Author [José Duarte <redacted>, n to skip]:  
Invalid author string. Must be in the format: John Smith <john@example.com>
```
If I try `Jose...` instead, it works though.

I didn't try other characters, but I'd suspect they also fail.

**Comment by dimbleby** [comment]
For me, é fails but é succeeds.  yuk.

I suspect those of us with ascii names are less motivated to figure out the difference between these things.  Merge request welcome, I imagine.

**Comment by jmg-duarte** [comment]
Could you share the code points? My computer doesn't catch a difference between them when pasted in Python:
```python
ord("é".encode("utf-8")) # first one
>>> b'\xc3\xa9'
ord("é".encode("utf-8")) # second one
>>> b'\xc3\xa9'
```

With the original regex:
https://github.com/python-poetry/poetry-core/blob/56c03764e1359615adfcacab12ababd7c57a0e90/src/poetry/core/packages/package.py#L38-L40

Pythex does catch the issue: https://pythex.org/?regex=(%3Fu)%5E(%3FP%3Cname%3E%5B-%20.%2C%5Cw%5Cd%27%E2%80%99%5C%22()%3A%26%5D%2B)(%3F%3A%20%3C(%3FP%3Cemail%3E.%2B%3F)%3E)%3F%24&test_string=Jose%CC%81%20Duarte%20%3Cjmg.duarte%40gmail.com%3E%0AJos%C3%A9%20Duarte%20%3Cjmg.duarte%40gmail.com%3E%0A&ignorecase=0&multiline=1&dotall=0&verbose=0

**Comment by dimbleby** [comment]
65 CC 81 vs C3 A9

github has preserved the difference so it's your copy-paste that is lossy

**Comment by jmg-duarte** [comment]
Thank you for sharing the codes!

> github has preserved the difference so it's your copy-paste that is lossy

This made no sense to me, when I pasted in Pythex it worked fine! So I was led down the rabbit hole and I found the issue:
MacOS uses (or at the very least used) a slightly different encoding (utf-8-mac). 
- https://gist.github.com/JamesChevalier/8448512
- https://apple.stackexchange.com/questions/346453/what-the-difference-and-usage-of-encodings-utf-8-and-utf-8-mac-in-iconv
- https://metacpan.org/pod/Encode::UTF8Mac

When I write "é" `C3A9` comes out and if I go through `poetry init` and re-write my name by hand, it works! However, the file where Poetry is getting that information from is probably encoded with the MacOS format. The simplest solution that will most likely always work is to just re-encode as UTF-8.

I'll take a better look later and submit a PR to fix this.
